plugin/src/main/groovy/com/novoda/staticanalysis/internal/lint/LintConfigurator.groovy

```diff
@@ -0,0 +1,64 @@
+package com.novoda.staticanalysis.internal.lint
+
+import com.novoda.staticanalysis.StaticAnalysisExtension
+import com.novoda.staticanalysis.internal.Configurator
+import com.novoda.staticanalysis.internal.Violations
+import org.gradle.api.NamedDomainObjectContainer
+import org.gradle.api.Project
+import org.gradle.api.Task
+
+class LintConfigurator implements Configurator {
+
+    private final Project project
+    private final Violations violations
+    private final Task evaluateViolations
+
+    static LintConfigurator create(Project project,
+                                   NamedDomainObjectContainer<Violations> violationsContainer,
+                                   Task evaluateViolations) {
+        Violations violations = violationsContainer.maybeCreate('Lint')
+        return new LintConfigurator(project, violations, evaluateViolations)
+    }
+
+    private LintConfigurator(Project project, Violations violations, Task evaluateViolations) {
+        this.project = project
+        this.violations = violations
+        this.evaluateViolations = evaluateViolations
+    }
+
+    @Override
+    void execute() {
+        project.extensions.findByType(StaticAnalysisExtension).ext."lintOptions" = { Closure config ->
+
+            if (!isAndroidProject(project)) {
+                return
+            }
+
+            project.extensions.findByName('android').lintOptions(config)
+
+            configureToolTask()
+        }
+    }
+
+    private void configureToolTask() {
+        // evaluate violations after lint
+        def collectViolations = createCollectViolationsTask(violations)
+        evaluateViolations.dependsOn collectViolations
+        collectViolations.dependsOn project.tasks['lint']
+    }
+
+    private CollectLintViolationsTask createCollectViolationsTask(Violations violations) {
+        def outputFolder = new File(project.projectDir, 'build/reports')
```

Comment:
As follow-up it would be good to resolve this path dynamically. Otherwise this will break in case the user configures another `xmlOutput`

Reply:
we should fix this now, as we shouldn't hardcode anything that can be configured. The only constraint we actually have is that we need to get this information after lint is configured, in order to read the correct path to the xml report. We have two ways of doing this as I see it: we get it from `lintOptions` or from the `lint` task.
If we assume that the user will configure lint through our plugin - effectively done at L36 - then we should read the value from the lint extension right afterwards, right?

Reply:
Also keep in mind that `lint` is run for all variants. Maybe we should have `includeVariants` here too. But in any case, should we also support multiple outputs?

Reply:
@mr-archano this PR is towards a feature branch. Not on develop. So I think we can just merge it without this? with a follow-up soon?

Reply:
@mr-archano just checked that, but the output path of `lintOptions` is null in case the user hasn't configured it. So we need to find another way. I agree with @tasomaniac to follow up on this as part of [this](https://novoda.atlassian.net/browse/PT-516) ticket.

Reply:
We agreed on for now to check whether the output path of `lintOptions` is configured or to fall back to the hardcoded one.

Reply:
@mr-archano done